android-json-form-wizard/src/main/java/com/vijay/jsonwizard/presenters/JsonFormFragmentPresenter.java

```diff
@@ -890,4 +893,12 @@ public boolean onMenuItemClick(MenuItem item) {
 
     return false;
   }
+
+  public void checkAndStopCountdownAlarm() {
+    // Check if alarm is ringing and stop
+    JSONObject step = getView().getStep(mStepName);
+    if(step.optJSONArray(JsonFormConstants.FIELDS).optJSONObject(0).has(JsonFormConstants.COUNTDOWN_TIME_VALUE)) {
```

Comment:
You can use `FormUtils#fields(JSONObject jsonForm, String step)` for this.

Since someone might embed the timer along with other widgets on one screen, you might want to iterate over the array this returns and test whether type is `JsonFormConstants.COUNTDOWN_TIME_VALUE`.

i.e. As it is it assumes only the timer widget exists on this page.

Reply:
Cool. Was thinking of the countdown timer as a single screen in terms of its use, but it makes sense to provide handling of instances where it's used in a screen with several other widgets
:heavy_check_mark: